README.md

```diff
@@ -1,11 +1,158 @@
+
+This [Jira epic](https://dhis2.atlassian.net/browse/DHIS2-14051) lists the app that moved to using the library, and what comes next.
+
+# Hooks for Calendar UI
+The library provides a hook `useDatePicker` that's consumed by the dhis/ui calendar components. There are two components that currently use it:
+- [Calendar](https://ui.dhis2.nu/components/calendar): a calendar component supporting non-Greogorian calendars
+- [CalendarInput](https://ui.dhis2.nu/components/calendar-input): a wrapper around the Calendar component to support the most common use case for a calendar when we want to display it next to an input.
+
+`useDatePicker` takes a [DatePickerOptions](https://github.com/dhis2/multi-calendar-dates/blob/multi-calendar-docs/src/hooks/useDatePicker.ts#L16) object, and returns [UseDatePickerReturn](https://github.com/dhis2/multi-calendar-dates/blob/multi-calendar-docs/src/hooks/useDatePicker.ts#LL28C13-L28C32) that contains information needed to render a UI component, i.e. the localised day names, and the weeks in the current view (month).
+
+```ts
+// the options passed to useDatePicker
+type DatePickerOptions = {
+    date: string
+    options: PickerOptions
+    onDateSelect: ({
+        calendarDate,
+        calendarDateString,
+    }: {
+        calendarDate: Temporal.ZonedDateTime
+        calendarDateString: string
+    }) => void
+}
+```
+
+```ts
+// the return type of the hook
+type UseDatePickerReturn = UseNavigationReturnType & {
+    weekDayLabels: string[]
+    calendarWeekDays: {
+        zdt: Temporal.ZonedDateTime
+        label: string | number
+        calendarDate: string
+        onClick: () => void
+        isSelected: boolean | undefined
+        isToday: boolean
+        isInCurrentMonth: boolean
+    }[][]
+}
+```
+
+# Periods and Dates helpers
+
+The library also provides helper methods to work with periods and dates in multiple calendars. This allows doing date arithmetic, comparisons in a standard timezone-safe way. It currently relies on the Temporal API.
+
+> The library is built on top of Temporal API, but this is an implementation detail, and one of the goals of the library is to hide that detail to allow replacing Temporal with a different implementation in the future, i.e. a Kotlin multi-platform implementation to achieve consistency between web, android and backend.
+
+## generateFixedPeriods
+
+- `generateFixedPeriods` returns the periods of a specific type in a specific year. This is used, for example in [Data Visualiser](https://github.com/dhis2/data-visualizer-app/pull/2233), to display the list of periods of different types (i.e. monthly, weekly, yearly) in a specific year, according to the user's calendar and locale.
+
+The method takes an `options` object of type `GeneratedPeriodParams` and returns an `Array<FixedPeriod>`.
```

Comment:
I like how you add the types in the doc here -- can you add one for `FixedPeriod`? 😊 (Maybe you omitted it because you thought it was too obvious hah)

Reply:
Ah yeah thanks for pointing that out, I guess I lost track between the description and the type 😅 

If other people are goldfish like me, then it might help to move it closer if possible 😅